mii/utils.py

```diff
@@ -209,6 +209,41 @@ def get_num_gpus(mii_configs):
     return num_gpus
 
 
+def get_provider_name(model_name, task):
+    if model_name == "gpt-neox":
+        provider = mii.constants.MODEL_PROVIDER_NAME_EA
+    elif ("bigscience/bloom" == model_name) or ("microsoft/bloom" in model_name):
+        provider = mii.constants.MODEL_PROVIDER_NAME_HF_LLM
+    elif task == mii.Tasks.TEXT2IMG:
+        provider = mii.constants.MODEL_PROVIDER_NAME_DIFFUSERS
+    else:
+        provider = mii.constants.MODEL_PROVIDER_NAME_HF
+    return provider
+
+
+def create_conversation(request, **kwargs):
+    if isinstance(request, dict):
+        assert 'text' in request and 'past_user_inputs' in request and 'generated_responses' in request, "Conversation requires 'text', 'past_user_inputs', and 'generated_responses' keys"
+        text = request['text']
+        conversation_id = request[
+            'conversation_id'] if 'conversation_id' in request else None
+        past_user_inputs = request['past_user_inputs']
+        generated_responses = request['generated_responses']
+
+    else:
+        text = getattr(request, 'text')
+        conversation_id = getattr(request, 'conversation_id')
+        past_user_inputs = getattr(request, 'past_user_inputs')
+        generated_responses = getattr(request, 'generated_responses')
+
+    conv = Conversation(text=text,
+                        conversation_id=conversation_id,
+                        past_user_inputs=past_user_inputs,
+                        generated_responses=generated_responses,
+                        **kwargs)
+    return conv
```

Comment:
I think this should be part of the `ConversationMethods` class rather than here in the utils.